cypress-io/cypress
# feat: reflect the GQL baseError state in App, not just launchpad

- Closes https://cypress-io.atlassian.net/browse/UNIFY-1595 and https://cypress-io.atlassian.net/browse/UNIFY-1440

### User facing changelog


- `baseError` state will be surfaced in the app now, as well as the launchpad
- a loading spinner will appear when the config has been saved and is reloading, or the config has been reloaded manually with the "try again" button from the error state.


### Additional details

This PR moves BaseError.vue into `@packages/frontend-shared` and adds a subscription to this state in the App. It also adds some more calls for `toApp()` around the same places we call `toLaunchpad()` to keep state in sync.

One side not is that without a valid config, we were not initially able to serve the application at all, which broke reloading behavior, so there are some changes to avoid serving a 500 with an error payload in that situation. With the 500 avoided, we still hit some problems loading the app with no config resolved, so I added defaults for `namespace` and `socketIoRoute` so that the front end can loaded/

### How has the user experience changed?

Previously the BaseError component was only in Launchpad and the browser App would not show these kinds of errors. Now they appear.

https://user-images.githubusercontent.com/8340719/166287918-502b9add-5a8a-4b88-a6d7-7f737986ee0c.mov

The 2nd half of the bug reported in [UNIFY-1440](https://cypress-io.atlassian.net/browse/UNIFY-1440) is also addressed by this PR, config values for project ID are now synced in both config and Dashboard Settings parts of the settings page:


https://user-images.githubusercontent.com/8340719/166306693-627285fa-13e5-4180-ada5-857dd1a8c60b.mov



### How to test manually

With the App and Launchpad open on a project, break the config file by setting an invalid value, or breaking the syntax of the the file. Errors should appear in both places. Refreshing should work in both places. "Try again" should work in both places. After fixing the config file, "try again" button should clear the error from both places.

When the App is open, it looks like config errors are the only thing that trigger a Base Error state, @tgriesser is this correct? I see some other things but they would only happen during migration. I've added tests related to the config state, if can think of other ways to get a baseError, I'll test them as well.

For UNIFY-1440 testing, change the `projectId` in the config whilst on the settings page and see that the new ID gets populated in both the config section and the Dashboard Settings section.

### PR Tasks


- [na] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [na] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?
